tools/rosgraph/src/rosgraph/xmlrpc.py

```diff
@@ -76,9 +76,32 @@ def isstring(s):
     except NameError:
         return isinstance(s, str)
 
-class SilenceableXMLRPCRequestHandler(SimpleXMLRPCRequestHandler):
 
-    protocol_version = 'HTTP/1.1'
+def check_kernel(major, minor):
+    """Small helper version to check that the current platform kernel is a
+    newer or the same as the input
+    """
+    release = platform.release().split('.')
+
+    platform_major = int(release[0])
+    platform_minor = int(release[1])
+
+    if platform_major > major:
+        return True
+    elif platform_major < major:
+        return False
+
+    if platform_minor > minor:
+        return True
+    elif platform_minor < minor:
+        return False
+
+    return True
+
+
+class SilenceableXMLRPCRequestHandler(SimpleXMLRPCRequestHandler):
+    if check_kernel(4,16):
```

Comment:
I think that this problem only happens on linux platform. I am not windows user but this will break the windows? I was thinking that it would be nice to specify the linux platform on this fix.

Reply:
checks specifically for linux now!